subsys/usb/device_next/class/usbd_uvc.c

```diff
@@ -0,0 +1,2273 @@
+			break;
+		}
+		if (ret == UVC_VBUF_DONE) {
+			LOG_DBG("Video buffer %p transferred, removing from the queue", vbuf);
+			k_fifo_get(&data->fifo_in, K_NO_WAIT);
+		}
+	}
+
+	/* Now the other contexts calling this function can access the fifo safely. */
+	LOG_DBG("Unlocking the UVC stream");
+	k_mutex_unlock(&data->mutex);
+}
+
+static void uvc_enable(struct usbd_class_data *const c_data)
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct uvc_data *data = dev->data;
+
+	__ASSERT_NO_MSG(atomic_test_bit(&data->state, UVC_STATE_INITIALIZED));
+
+	atomic_set_bit(&data->state, UVC_STATE_ENABLED);
+
+	/* Catch-up with buffers that might have been delayed */
+	uvc_flush_queue(dev);
+}
+
+static void uvc_disable(struct usbd_class_data *const c_data)
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct uvc_data *data = dev->data;
+
+	__ASSERT_NO_MSG(atomic_test_bit(&data->state, UVC_STATE_INITIALIZED));
+
+	atomic_clear_bit(&data->state, UVC_STATE_ENABLED);
+}
+
+static void uvc_update(struct usbd_class_data *const c_data, uint8_t iface, uint8_t alternate)
+{
+	LOG_DBG("Select alternate %u for interface %u", alternate, iface);
+}
+
+static const struct usbd_class_api uvc_class_api = {
+	.enable = uvc_enable,
+	.disable = uvc_disable,
+	.request = uvc_request,
+	.update = uvc_update,
+	.control_to_host = uvc_control_to_host,
+	.control_to_dev = uvc_control_to_dev,
+	.init = uvc_init,
+	.get_desc = uvc_get_desc,
+};
+
+/* UVC video API */
+
+static int uvc_enqueue(const struct device *dev, struct video_buffer *vbuf)
+{
+	struct uvc_data *data = dev->data;
+
+	k_fifo_put(&data->fifo_in, vbuf);
+	uvc_flush_queue(dev);
+
+	return 0;
+}
+
+static int uvc_dequeue(const struct device *dev, struct video_buffer **vbuf, k_timeout_t timeout)
+{
+	struct uvc_data *data = dev->data;
+
+	*vbuf = k_fifo_get(&data->fifo_out, timeout);
+	if (*vbuf == NULL) {
+		return -EAGAIN;
+	}
+
+	return 0;
+}
+
+static int uvc_get_format(const struct device *dev, struct video_format *fmt)
+{
+	struct uvc_data *data = dev->data;
+	struct video_format tmp_fmt = {0};
+	int ret;
+
+	__ASSERT_NO_MSG(data->video_dev != NULL);
+
+	if (!atomic_test_bit(&data->state, UVC_STATE_ENABLED) ||
+	    !atomic_test_bit(&data->state, UVC_STATE_STREAM_READY)) {
+		return -EAGAIN;
+	}
+
+	LOG_DBG("Querying the format from %s", data->video_dev->name);
+
+	tmp_fmt.type = VIDEO_BUF_TYPE_OUTPUT;
+
+	ret = video_get_format(data->video_dev, &tmp_fmt);
+	if (ret != 0) {
+		return ret;
+	}
+
+	*fmt = tmp_fmt;
+
+	return 0;
+}
+
+static int uvc_set_stream(const struct device *dev, bool enable, enum video_buf_type type)
+{
+	struct uvc_data *data = dev->data;
+
+	if (enable) {
+		atomic_clear_bit(&data->state, UVC_STATE_PAUSED);
+		uvc_flush_queue(dev);
+	} else {
+		atomic_set_bit(&data->state, UVC_STATE_PAUSED);
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_POLL
+static int uvc_set_signal(const struct device *dev, struct k_poll_signal *sig)
```

Comment:
Have you tried building it with CONFIG_POOL disabled?

Reply:
I tested with `CONFIG_POLL=n` and the sample has lacking definitions of `k_poll()` related functions.
After commenting them out from the sample, UVC builds.
I can modify the sample so that `CONFIG_POLL=y` is not required if needed.